modules/default/compliments/compliments.js

```diff
@@ -38,11 +40,40 @@ Module.register("compliments", {
 			this.config.compliments = JSON.parse(response);
 			this.updateDom();
 		}
+		let minute_sync_delay = 1;
+		// loop thru all the configured when events
+		for (let m of Object.keys(this.config.compliments)) {
+			// if one contains a space, its a cron entry
+			if (m.includes(" ")) {
+				// we need to synch our interval cycle to the minute
+				minute_sync_delay = (60 - (moment().second())) * 1000;
+				break;
+			}
+		}
+		// Schedule update timer. sync to the minute start (if needed), so minute based events happen on the minute start
+		setTimeout(() => {
+			setInterval(() => {
+				this.updateDom(this.config.fadeSpeed);
+			}, this.config.updateInterval); },
+		minute_sync_delay);
+	},
+
+	/**
+	 * @param {string} cronExpression The cron expression. See https://croner.56k.guru/usage/pattern/
+	 * @param {Date} [timestamp] The timestamp to check. Defaults to the current time.
+	 * @returns {number} The number of seconds until the next cron run.
+	 */
+	getSecondsUntilNextCronRun (cronExpression, timestamp = new Date()) {
+		// Required for seconds precision
+		const adjustedTimestamp = new Date(timestamp.getTime() - 1000);
 
-		// Schedule update timer.
-		setInterval(() => {
-			this.updateDom(this.config.fadeSpeed);
-		}, this.config.updateInterval);
+		// https://www.npmjs.com/package/croner
+		/*eslint-disable-next-line*/
```

Comment:
why this disable?

Reply:
because the linter fails that line cause there is no matching function definition (require) as the script is loaded thru getScripts

Reply:
```sh
/home/sam/MagicMirror/modules/default/compliments/compliments.js
  78:23  error  'Cron' is not defined  no-undef
```

Reply:
https://gitkraken.dev/link/drafts/c673b0d9-8e28-4640-8fe5-7e22673262d4?type=suggested_pr_change&prEntityId=WyJnaXRodWIiLCJwciIsIjEiLCIiLCJQUl9rd0RPQVFNOHU4NXpiTl8tIl0=

Reply:
Then add a global to the top of the file (like we do for other libs):

/* global Cron */

Reply:
ok, that makes no sense.. but it worked..